clientv3/client_test.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2016 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package clientv3
+
+import (
+	"testing"
+	"time"
+)
+
+func TestDialTimeout(t *testing.T) {
+	donec := make(chan struct{}, 1)
+	go func() {
+		cfg := Config{
+			Endpoints:   []string{"localhost:12345"},
+			DialTimeout: 2 * time.Second}
+		if c, err := New(cfg); c != nil || err == nil {
```

Comment:
just setup a plain tcp on the other side that blocks on accept? also we should test the connection refuse case too, right?